services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy

```diff
@@ -184,6 +184,64 @@ class IntentionalRerouteSpec extends BaseSpecification {
         flowHelper.deleteFlow(flow.id)
     }
 
+    def "Should be able to reroute to a path with not enough bandwidth available in case ignoreBandwidth=true"() {
+        given: "A flow with alternate paths available"
+        def switches = topology.getActiveSwitches()
+        List<List<PathNode>> allPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.unique { it.sort() }.find { Switch src, Switch dst ->
+            allPaths = database.getPaths(src.dpId, dst.dpId)*.path
+            allPaths.size() > 1
+        } ?: assumeTrue("No suiting switches found", false)
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = 10000
+        flow.ignoreBandwidth = true
+        flowHelper.addFlow(flow)
+        def currentPath = PathHelper.convert(northbound.getFlowPath(flow.id))
+
+        when: "Make the current path less preferable than alternatives"
+        def alternativePaths = allPaths.findAll { it != currentPath }
+        alternativePaths.each { pathHelper.makePathMorePreferable(it, currentPath) }
+
+        and: "Make all alternative paths to have not enough bandwidth to handle the flow"
+        def currentIsls = pathHelper.getInvolvedIsls(currentPath)
+        def changedIsls = alternativePaths.collect { altPath ->
+            def thinIsl = pathHelper.getInvolvedIsls(altPath).find {
+                !currentIsls.contains(it) && !currentIsls.contains(it.reversed)
+            }
+            def newBw = flow.maximumBandwidth - 1
+            [thinIsl, thinIsl.reversed].each {
+                database.updateLinkMaxBandwidth(it, newBw)
+                database.updateLinkAvailableBandwidth(it, newBw)
+            }
+            thinIsl
+        }
+
+        and: "Init a reroute to a more preferable path"
+        def rerouteResponse = northbound.rerouteFlow(flow.id)
+
+        then: "The flow is rerouted because ignoreBandwidth:true"
+        int seqId = 0
+
+        rerouteResponse.rerouted
+        rerouteResponse.path.path != currentPath
+        rerouteResponse.path.path.each { assert it.seqId == seqId++ }
+
+        PathHelper.convert(northbound.getFlowPath(flow.id)) != currentPath
+        Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.UP }
+        def cPath = PathHelper.convert(northbound.getFlowPath(flow.id))
+        def allIsl = northbound.getAllLinks()
+        def firstIsl = pathHelper.getInvolvedIsls(cPath).collect { islUtils.getIslInfo(allIsl, it).get() }[0]
+        firstIsl.availableBandwidth < flow.maximumBandwidth
```

Comment:
Don't understand what is being verified here. You should: take isl bw before reroute, and verify that it is the same after reroute. The `firstIsl.availableBandwidth < flow.maximumBandwidth` basically does nothing. The `firstIsl.availableBandwidth` may be 0 or even drop into negative value and your test will pass. This is an autopass verification.

Reply:
I will compare it against the newBw variable
just to avoid introducing new variable
it should be accepted here, cause we just want to be sure that available bandwidth was not changed after rerouting